src/atomic.rs

```diff
@@ -115,7 +115,7 @@ unsafe impl<T: Send + Sync> Sync for Atomic<T> {}
 impl<T> Atomic<T> {
     /// Returns a new atomic pointer pointing to the tagged pointer `data`.
     fn from_data(data: usize) -> Self {
-        Atomic {
+        Self {
```

Comment:
> I find browsing documentation much easier when types are explicitly spelled out in signatures rather than abbreviated with `Self`.

I personally disagree here; if `Self` is more difficult to read in the documentation, maybe it's a bug in `rustdoc`..  But more importantly,

>  The standard library apparently uses the same approach.

I'm fully willing to follow the standard library's approach.  Tomorrow I'll rename `Self` into the actual type names in the public signatures.

Reply:
> I personally disagree here; if `Self` is more difficult to read in the documentation, maybe it's a bug in `rustdoc`

Great idea! I'll open an issue on this in the rust repository.